hpx/runtime/components/client_base.hpp

```diff
@@ -289,11 +289,18 @@ namespace hpx { namespace components
         }
 
         ///////////////////////////////////////////////////////////////////////
+#if defined(HPX_HAVE_COMPONENT_GET_GID_COMPATIBILITY)
         naming::id_type const & get_gid() const
         {
             return gid_.get();
         }
 
+#endif
+        naming::id_type const & get_id() const
+        {
+            return gid_.get();
+        }
+
```

Comment:
This is completely orthogonal to getting rid of id_type. We have them right now, that's a fact, this PR is dealing with an API related to id_type, not getting rid of it. Unfortunately, `get_id` in client_base could return either a managed or unmanaged `id_type`. But that's probably OK for now.